core/store/src/trie/update.rs

```diff
@@ -86,14 +73,28 @@ impl TrieUpdate {
                 return Ok(data.as_ref().map(TrieUpdateValuePtr::MemoryRef));
             }
         }
-
-        self.trie.get_ref(&self.root, &key).map(|option| {
+        self.trie.get_ref(&key).map(|option| {
             option.map(|ValueRef { length, hash }| {
                 TrieUpdateValuePtr::HashAndSize(&self.trie, length, hash)
             })
         })
     }
 
+    pub fn get(&self, key: &TrieKey) -> Result<Option<Vec<u8>>, StorageError> {
+        self.get_impl(&key.to_vec())
+    }
+
+    fn get_impl(&self, key: &[u8]) -> Result<Option<Vec<u8>>, StorageError> {
```

Comment:
Hm, was this some partial attempt at refactoring dup with `get_ref`? Ultra-minor, but I'd probably reverted this change, to keep `get_ref` and `get` the same (or pushed the refactor to completion with `fn get_from_mem(&self, key: &[u8]) -> Option<Option<&[u8]>>`)

Reply:
That’s not quite it.  I have another pending change which introduces TrieAccess trait and this should go into that PR.